src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsJavaDocsTest.java

```diff
@@ -699,7 +699,7 @@ private static boolean isSetterMethod(DetailAST ast) {
         private static String getJavaDocText(DetailAST node) {
             final String text = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<document>\n"
                     + node.getFirstChild().getText().replaceAll("(^|\\r?\\n)\\s*\\* ?", "\n")
-                            .replaceAll("\\n@noinspection.*\\r?\\n", "\n")
+                            .replaceAll("\\n?@noinspection.*\\r?\\n[^@]*", "\n")
```

Comment:
Required to not add `noinspectionreason` content to xdocs

Reply:
Build will fail without this change: https://github.com/checkstyle/checkstyle/pull/11880/checks?check_run_id=7272320294